python/cudf/cudf/core/timestamp.py

```diff
@@ -0,0 +1,389 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+from __future__ import annotations
+
+from datetime import datetime, timedelta
+
+import numpy as np
+import pandas as pd
+
+from cudf.core.scalar import Scalar
+
+
+class Timestamp(Scalar):
```

Comment:
I wonder if would be better to first get the implementations of `cudf.Timestamp` and `cudf.Timedelta` in and track their performance implications before replacing them throughout cudf (eg. [here](https://github.com/rapidsai/cudf/blob/ac42bc870a65d807784cae63e25b9e9ca788eb23/python/cudf/cudf/core/column/datetime.py#L574))